extensions/replication/utils/SetupReplication.js

```diff
@@ -163,39 +193,33 @@ class SetupReplication extends BackbeatTask {
         // Does the bucket exist and is it reachable?
         const bucket = where === 'source' ? this._sourceBucket :
             this._targetBucket;
-        this._s3Clients[where].headBucket({ Bucket: bucket }, err => {
-            if (err) {
+        (async () => {
+            try {
+                const command = new HeadBucketCommand({ Bucket: bucket });
+                await this._s3Clients[where].send(command);
+                cb();
```

Comment:
do not invoke callback from within try block : use promise.then() or callbackify()
(same for other calls in this file)